newIDE/app/src/Utils/GDevelopServices/Asset.js

```diff
@@ -177,25 +179,38 @@ export const listAllAssets = ({
     });
 };
 
-export const getAsset = (
+export const getPublicAsset = async (
   assetShortHeader: AssetShortHeader,
   { environment }: {| environment: Environment |}
 ): Promise<Asset> => {
-  return client
-    .get(`/asset/${assetShortHeader.id}`, {
-      params: {
-        environment,
-      },
-    })
-    .then(response => response.data)
-    .then(({ assetUrl }) => {
-      if (!assetUrl) {
-        throw new Error('Unexpected response from the asset endpoint.');
-      }
+  const response = await client.get(`/asset/${assetShortHeader.id}`, {
+    params: {
+      environment,
+    },
+  });
+  if (!response.data.assetUrl) {
+    throw new Error('Unexpected response from the asset endpoint.');
+  }
 
-      return client.get(assetUrl);
-    })
-    .then(response => response.data);
+  const assetResponse = await client.get(response.data.assetUrl);
+  return assetResponse.data;
+};
+
+export const getPrivateAsset = async (
+  assetShortHeader: AssetShortHeader,
+  authorizationToken: string,
+  { environment }: {| environment: Environment |}
+): Promise<Asset> => {
+  const assetPackId = assetShortHeader.assetPackId;
+  if (!assetPackId) {
+    throw new Error('Asset is not part of a private asset pack.');
```

Comment:
Reading the code this just tells me "the asset is not part of an asset pack that has an ID".
The fact that it means that it's private is just a coincidence, true for now but that could be false in the future, which is:
"It turns out that only private asset packs have ids".

So we should rename "assetPackId" to "privateAssetPackId" or, maybe preferably, just change the error message to:
```suggestion
    throw new Error('Asset is not part of an asset pack that can be accessed with an id.');
```